### DISCUSSION TITLE
Local function overloads

### DISCUSSION THREAD

**Discussion by rbwhitaker** [dotnet/csharplang#4224] [Language Ideas] [answered]
I'd like to suggest the idea of allowing local functions to include overloads. I don't see a similar proposal anywhere, but especially with a small program composed only of top-level statements and some methods, it seems like it could be useful. I was just trying to write a small console app with a method to get a number from the user, and also to get a number from the user within a specific range:

```cs
using System;

// Some random code here.

int AskForNumber(string text)
{
    Console.Write("text");
    return Convert.ToInt32(Console.ReadLine());
}

int AskForNumber(string text, int min, int max)
{
    while (true)
    {
        int number = AskForNumber(text);
        if (number >= min && number<= max) return number;
    }
}
```

Unfortunately, that doesn't work in C# 9 because "CS0128: A local variable or function named 'AskForNumber' is already defined in this scope". Admittedly, overloads of local functions are likely pretty rare. But I think with top-level statements, they're probably about to become more common.

**Comment by jnm2** [comment, marked as answer]
Duplicate of https://github.com/dotnet/csharplang/discussions/3112

**Reply by rbwhitaker** [reply]
I agree. Thanks for finding that.
